op-node/rollup/derive/span_batch.go

```diff
@@ -397,13 +397,16 @@ func (b *SpanBatch) ToRawSpanBatch(originChangedBit uint, genesisTimestamp uint6
 	return &raw, nil
 }
 
-// GetSingularBatches returns a list of SingularBatches that converted from spanBatchElements.
+// GetSingularBatches converts spanBatchElements after L2 safe head to SingularBatches.
 // Since spanBatchElement does not contain EpochHash, set EpochHash from the given L1 blocks.
 // The result SingularBatches do not contain ParentHash yet. It must be set by BatchQueue.
-func (b *SpanBatch) GetSingularBatches(l1Origins []eth.L1BlockRef) ([]*SingularBatch, error) {
+func (b *SpanBatch) GetSingularBatches(l1Origins []eth.L1BlockRef, l2SafeHead eth.L2BlockRef) ([]*SingularBatch, error) {
```

Comment:
Can we update method name? It now has additional responsibility.
Or we can leave `GetSingularBatches` and implement new method for this update

Reply:
Idea: use lambda which accepts batch and returns bool. If lambda is true, append to return list.

Like

```go
func (b *SpanBatch) GetSingularBatches(l1Origins []eth.L1BlockRef, query func(batch) bool) {

    for _, batch := rane b.batches {
         if !query(batch) {
             continue
         }
         ...
    }
}